File: python/palindrom.py
def main():
    n = int(input('Podaj liczbę n: '))
    p = int(input('Podaj podstawę p: '))

    if n < 0:
        print('Niepoprawne dane (liczba n jest ujemna)')
        return
    if p <= 1:
        print('Niepoprawne dane (podstawa p mniejsza od 1)')
        return

    n_copy = n
    n_reverse = 0
    
    while n > 0:
        n_reverse = n_reverse * p + n % p
        n //= p

    print('Palindrom' if n_copy == n_reverse else 'Nie jest palindromem')


if __name__ == '__main__':
    main()

File: c/palindrom.c
#include <stdio.h>

int main() {
    int n, p, n_copy, n_reverse = 0;

    printf("Podaj liczbe n: ");
    scanf("%d", &n);
    printf("Podaj podstawę p: ");
    scanf("%d", &p);
   
    if (n < 0) {
        printf("Niepoprawne dane (liczba n jest ujemna)\n");
        return 0;
    }
    if (p <= 1) {
        printf("Niepoprawne dane (podstawa pjest mniejsza od 1)\n");
        return 0;
    }

    n_copy = n;

    while (n > 0) {
        n_reverse = n_reverse * p + n % p;
        n /= p;
    }

    printf((n_copy == n_reverse) ? "Palindrom\n" : "Nie jest palindromem\n");

    return 0;
}